src/lib/operators/join_index.cpp

```diff
@@ -335,18 +446,23 @@ void JoinIndex::_write_output_segments(Segments& output_segments, const std::sha
             input_table->get_chunk(ChunkID{0})->get_segment(column_id));
 
         // de-reference to the correct RowID so the output can be used in a Multi Join
-        for (const auto& row : *pos_list) {
+        for (auto pos_list_offset = ChunkOffset{0}; pos_list_offset < pos_list->size(); ++pos_list_offset) {
+          const auto& row = (*pos_list)[pos_list_offset];
```

Comment:
Dereferencing `pos_list` outside of the loop might save some performance.

Reply:
I changed this in ebf28ddb0671b708f6e605b44e8cbee97f61b4b1 but [this experiment](http://quick-bench.com/wL4aoNRhIBiiOClfd7Jqwofir14) leads me to the assumption that this won't improve the performance.

Reply:
Thanks for setting up the experiment. Apparently, the compiler can optimize that. For GCC it actually is 10% faster. Markus pointed out to me that it the current solution in ebf28dd does not make any difference. Could you just use the previous version, dereferencing the PosList in the loop? Sorry!